src/main/java/net/snowflake/client/jdbc/SnowflakeFileTransferAgent.java

```diff
@@ -1950,11 +1951,14 @@ static Set<String> expandFileNames(String[] filePathList, String queryId)
             && injectedFileTransferException instanceof Exception) {
           throw (Exception) SnowflakeFileTransferAgent.injectedFileTransferException;
         }
-
         // The following currently ignore sub directories
-        for (Object file :
-            FileUtils.listFiles(dir, new WildcardFileFilter(entry.getValue()), null)) {
-          result.add(((java.io.File) file).getCanonicalPath());
+        File[] filesMathingPattern =
+            dir.listFiles((FileFilter) new WildcardFileFilter(entry.getValue()));
+        if (filesMathingPattern == null) {
```

Comment:
In what case we can get null?

Reply:
`listFiles` javadoc says:
```
Returns null if this abstract pathname does not denote a directory, or if an I/ O error occurs.
```

Probably it should not happen but let's play safe